### PR TITLE
Whats new notification

### LINKED ISSUE #33230: What's new notification frontend
Epic: [Provide a heads up when Metabase is upgraded](https://github.com/metabase/metabase/issues/33165)

From the technical point of view, this is how we'll decide if we need to show the notification:
- get the versions list from version-info
- we filter out all versions < of the last acknowledged (if not null)
- we filter out all versions > of the current version
- from the remaining, we get the last version that has a releaseNotesUrl and link to that page

# Testing plan:

## Dimensions:
- is the app in a full embed
- last acknowledged version (null or present)
- presence of a releaseNotesUrl before, inside, and after the range described above

## Actions
- dismiss notification
- click `What's new?` link
- ignore notification and change page/reload page

### Unit/Integration

- [x] if the app is in full embed context => nothing is shown
- [x] links have `_target="blank"` attribute
- [x] if a releaseNotesUrl is present in the range (last ack, current] => show the banner
- [x] if two releaseNotesUrl are present, the last one (more recent) is used
- [x] if the releaseNotesUrl is present but on a version bigger than the current, it is ignored
- [x] if the releaseNotesUrl is present, but on a version already acknowledged, it is ignored

### E2E
- [x] new major update flow
  - user opens metabase and notification shows up
  - user refreshes the page -> notification still there
  - user dismiss the notification -> notification hides
  - user refreshes the page -> notification not shown
- [x] multi user support
  - user A logins
  - user A dismiss notification
  - user B logins
  - -> should see notification

### PR DESCRIPTION
**Pull request by npretto**
Epic: https://github.com/metabase/metabase/issues/33165

Closes https://github.com/metabase/metabase/issues/33230

Product doc: https://www.notion.so/metabase/Provide-a-heads-up-when-Metabase-is-upgraded-with-what-s-new-44a28b55624f48e28dc3f616ce7cda0f

This PR adds a notification that links to the release notes.
The notification is shown if:
- the current version is present in `version-info` (this excludes snapshots, release candidates and betas)
- there's a version in `version-info` that:
  - has a announcement_url (this will be added only to major/golden releases)
  - is > than the last acknowledged version
  - is <= the current version
  
The notification is dismissible per user and will not be dismissed automatically.
It should not cover any content in the nav bar.

## How to test locally with cypress:
This features requires the current version to be released (to be in version-info), when testing/developing locally this is never the case as a special version name is used.

I created a cypress test that sets up the required settings so that we can test it locally, to use it we need to:
- run metabase locally on this branch (`yarn dev`) in a terminal
- edit the file `e2e/test/scenarios/onboarding/navbar/whats-new.cy.spec.js` and replace `it.skip` with `it.only` on line 27
- run cypress (`yarn test-cypress-open`) in another terminal (this will take some time)
- search for the correct test by filtering for `what`, you should see the following:
 
image

- click on "whats-new.cy"

- cypress should run the test and show you metabase's home with the notification showing up

You should be able to dismiss the notification, and the link should link to `metabase.com/releases/48` (which is 404, that's ok)

--- 
 

image

### CODE REVIEW

File: frontend/src/metabase/nav/components/WhatsNewNotification/WhatsNewNotification.tsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  if (!url) {
+    return null;
+  }
+  return (
+    <Paper my="lg" mx="auto" p="md" shadow="md" withBorder w={244}>
+      <Stack spacing="sm">
+        <Flex justify="space-between">
+          <Sparkles color={color("brand")} />
+          <IconButtonWrapper onClick={dimiss}>
+            <Icon name="close" color={color("bg-dark")} />
```
Comment (kdoh) [inline frontend/src/metabase/nav/components/WhatsNewNotification/WhatsNewNotification.tsx:52]: Let's have this darken on hover to make it feel interactive. We can use `text-medium` for the hover state color (I know the names are silly, we're working on it).

**Review by kdoh — CHANGES_REQUESTED** [review]
One minor thing about the icon hover state. I wasn't able to get cypress working locally to test the dismiss and appearance logic it bombed out without a lot of info why, so I just commented out line 44 to see the prompt in my local branch.

**Review by kdoh — APPROVED** [review]
LGTM. 👍

**Review by ranquild — APPROVED** [review]

File: frontend/src/metabase/lib/utils.js [hunk 2]
```diff
@@ -135,36 +135,58 @@ const MetabaseUtils = {
...
-        .split(/[.-]*([0-9]+)[.-]*/)
-        .filter(c => c)
-        // ["1", "2", "3", "beta", "1"]
-        .map(c => SPECIAL_COMPONENTS[c] || parseInt(c, 10));
-    // [1, 2, 3, -2, 1]
-
-    const aComponents = getComponents(aVersion);
-    const bComponents = getComponents(bVersion);
+    const regex =
+      /v?(?<ossOrEE>\d+)\.?(?<major>\d+)?\.?(?<minor>\d+)?\.?(?<patch>\d+)?-?(?<label>\D+)?(?<build>\d+)?/;
```
Comment (iethree) [inline frontend/src/metabase/lib/utils.js:153]: this is clever, but we should consider just using the battle-tested

https://www.npmjs.com/package/semver

Comment (npretto) [reply frontend/src/metabase/lib/utils.js:153]: @iethree We tried to use that but we're not really 100% semver standard compatible :( 
See https://github.com/metabase/metabase/pull/33671#issuecomment-1705431936

**Pull request merged**